utils/mono/mono.go

```diff
@@ -7,8 +7,12 @@ var (
 	epochNano = epoch.UnixNano()
 )
 
+func FromTime(t time.Time) time.Time {
+	return epoch.Add(t.Sub(epoch))
+}
+
 func Now() time.Time {
-	return epoch.Add(time.Since(epoch))
+	return FromTime(time.Now())
```

Comment:
This could make it non-monotonic if `now` changes? i. e. taking `Now()` is not monotonic.

Reply:
to correct myself here, @paulwe set me right, `time.Now()` has both wall and monotonic time and `t.Sub` in `FromTime` will use the monotonic time.